waspc/src/Analyzer/Parser/Util.hs

```diff
@@ -0,0 +1,66 @@
+module Analyzer.Parser.Util
+  ( ParserState (..),
+    initialState,
+    Parser,
+    updatePosition,
+    putInput,
+    setStartCode,
+    ParserInput,
+  )
+where
+
+import Analyzer.Parser.ParseError (ParseError)
+import Analyzer.Parser.Token (SourcePosition (..))
+import Control.Monad.Trans.Except (Except)
+import Control.Monad.Trans.State.Lazy (StateT, get, modify)
+import Data.Word (Word8)
+
+-- | Tracks state of the parser, which is
+--   - Current line/column position
+--   - Current input to the lexer
```

Comment:
This comment now feels redundant, all the information can be directly read from the definition of ParserState.